roofit/roofitcore/src/RooAddPdf.cxx

```diff
@@ -428,51 +379,42 @@ RooAddPdf::CacheElem* RooAddPdf::getProjCache(const RooArgSet* nset, const RooAr
   // *** PART 1 : Create supplemental normalization list ***
 
   // Retrieve the combined set of dependents of this PDF ;
-  RooArgSet *fullDepList = getObservables(nset) ;
+  auto fullDepList = std::unique_ptr<RooArgSet>{getObservables(nset)} ;
```

Comment:
Hi, 
Since you are moving to use unique pointer, would not be better to use instead the new functions which do not build the set, but they fill it::
```
RooArgSet fullDepList;
getObservables(nset, fullDepList);
```

Reply:
Good idea! I'll make these changes in a separate PR that uses the new functions also in more new places than RooAddPdf.